flink-connectors/flink-connector-kinesis/src/test/java/org/apache/flink/streaming/connectors/kinesis/FlinkKinesisConsumerTest.java

```diff
@@ -952,27 +965,33 @@ public void emitWatermark(Watermark mark) {
 
 		// trigger sync
 		testHarness.setProcessingTime(testHarness.getProcessingTime() + 1);
-		TestWatermarkTracker.assertSingleWatermark(-4);
+		TestWatermarkTracker.assertGlobalWatermark(-4);
 
 		final long record2 = record1 + (watermarkSyncInterval * 3) + 1;
 		shard1.put(Long.toString(record2));
 
-		// TODO: check for record received instead
-		Thread.sleep(100);
+		// wait for the record to be buffered in the emitter
+		final RecordEmitter<?> emitter = org.powermock.reflect.Whitebox.getInternalState(fetcher, "recordEmitter");
```

Comment:
Using `Whitebox` usually indicates that the class under test is not well suited for testing. Maybe it can be refactored.

Reply:
Yes, that's more of a philosophical discussion outside of the issue at hand. Whitebox avoids extra code to be added to the class under test purely for the purpose of assertions in the test case. The other side of the argument is usually that tests can break more easily because there isn't a compile time check for such access. I think using Whitebox within the same module is fine because one would always run the tests while making potentially breaking modifications.